scripts/generate_can.py

```diff
@@ -15,290 +15,297 @@
 import generate_can_enums
 import generate_can_wrapper
 
-supported_years = ['2021', '2022']
 
-HWBRIDGE_ROOT_PATH = os.getcwd()
+# https://github.com/eerimoq/cantools/blob/871581b57785fdbd79c118878f9b9c148984963c/cantools/database/can/c_source.py#L762
+def camel_to_snake_case(value):
+    value = re.sub(r'(.)([A-Z][a-z]+)', r'\1_\2', value)
+    value = re.sub(r'(_+)', '_', value)
+    value = re.sub(r'([a-z0-9])([A-Z])', r'\1_\2', value).lower()
+    value = re.sub(r'[^a-zA-Z0-9]', '_', value)
```

Comment:
since these regex are reused, it would be more efficient to cache them using `re.compile`

Reply:
> It is important to note that most regular expression operations are available as module-level functions and methods on [compiled regular expressions](https://docs.python.org/3/library/re.html#re-objects). The functions are shortcuts that don’t require you to compile a regex object first, but miss some fine-tuning parameters.

Look at https://docs.python.org/3/library/re.html#re.Pattern.sub